packages/task/src/browser/task-service.ts

```diff
@@ -378,8 +403,57 @@ export class TaskService implements TaskConfigurationClient {
         this.logger.debug(`Task created. Task id: ${taskInfo.taskId}`);
 
         // open terminal widget if the task is based on a terminal process (type: shell)
+        // or attach to process output processor if a raw process (type: process)
         if (taskInfo.terminalId !== undefined) {
-            this.attach(taskInfo.terminalId, taskInfo.taskId);
```

Comment:
Now that I see this code, I am certain that this is something I missed when trying to align `shell` and `process` tasks with VS Code.

Theia doesn't prompt a widget with the process' output for a `process`-type task, but VS Code does. And since all actual processes started with either task type is now a `TerminalProcess`, both should open a terminal in the frontend.

So this makes me wonder how to handle the branching you added to split between attaching a shell task and attaching a process task.